src/scanner.c

```diff
@@ -135,9 +139,11 @@ void            SyntaxError (
         Pr( "%s", (Int)STATE(Input)->line, 0L );
 
         /* print a '^' pointing to the current position                        */
-        for ( i = 0; i < STATE(In) - STATE(Input)->line - 1; i++ ) {
-          if ( STATE(Input)->line[i] == '\t' )  Pr("\t",0L,0L);
-          else  Pr(" ",0L,0L);
+        Int pos = GetLinePosition();
```

Comment:
I like the new `GetLinePosition`, even without the fix, because it helps to make the code more understandable. Nice :-)